pkg/networkaware/networkoverhead/networkoverhead_test.go

```diff
@@ -0,0 +1,1546 @@
+			nodes: nodes,
+			want:  nil,
+			wantedScoresBefore: framework.NodeScoreList{
+				framework.NodeScore{Name: nodes[0].Name, Score: 0},
+				framework.NodeScore{Name: nodes[1].Name, Score: 0},
+				framework.NodeScore{Name: nodes[2].Name, Score: 0},
+				framework.NodeScore{Name: nodes[3].Name, Score: 0},
+				framework.NodeScore{Name: nodes[4].Name, Score: 0},
+				framework.NodeScore{Name: nodes[5].Name, Score: 0},
+				framework.NodeScore{Name: nodes[6].Name, Score: 0},
+				framework.NodeScore{Name: nodes[7].Name, Score: 0},
+			},
+			wantedScoresAfter: framework.NodeScoreList{
+				framework.NodeScore{Name: nodes[0].Name, Score: 0},
+				framework.NodeScore{Name: nodes[1].Name, Score: 0},
+				framework.NodeScore{Name: nodes[2].Name, Score: 0},
+				framework.NodeScore{Name: nodes[3].Name, Score: 0},
+				framework.NodeScore{Name: nodes[4].Name, Score: 0},
+				framework.NodeScore{Name: nodes[5].Name, Score: 0},
+				framework.NodeScore{Name: nodes[6].Name, Score: 0},
+				framework.NodeScore{Name: nodes[7].Name, Score: 0},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// init listers
+			agClient := agfake.NewSimpleClientset()
+			ntClient := ntfake.NewSimpleClientset()
+
+			fakeAgInformer := aginformers.NewSharedInformerFactory(agClient, 0).Appgroup().V1alpha1().AppGroups()
+			fakeNTInformer := ntinformers.NewSharedInformerFactory(ntClient, 0).Networktopology().V1alpha1().NetworkTopologies()
+
+			// add CRDs
+			_ = fakeAgInformer.Informer().GetStore().Add(tt.appGroup)
+			_ = fakeNTInformer.Informer().GetStore().Add(tt.networkTopology)
+
+			agLister := fakeAgInformer.Lister()
+			ntLister := fakeNTInformer.Lister()
+
+			ctx := context.Background()
+			cs := testClientSet.NewSimpleClientset()
+
+			informerFactory := informers.NewSharedInformerFactory(cs, 0)
+
+			snapshot := newTestSharedLister(nil, tt.nodes)
+
+			podInformer := informerFactory.Core().V1().Pods()
+			podLister := podInformer.Lister()
+			informerFactory.Start(ctx.Done())
+
+			for _, p := range tt.pods {
+				_, err := cs.CoreV1().Pods("default").Create(ctx, p, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Failed to create Workload %q: %v", p.Name, err)
+				}
+				t.Logf("Workload %v created  \n", p.Name)
+			}
+
+			registeredPlugins := []st.RegisterPluginFunc{
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+			}
+
+			fh, _ := st.NewFramework(registeredPlugins, "default-scheduler", runtime.WithClientSet(cs),
+				runtime.WithInformerFactory(informerFactory), runtime.WithSnapshotSharedLister(snapshot))
+
+			pl := &NetworkOverhead{
+				handle:      fh,
+				agLister:    agLister,
+				podLister:   podLister,
+				ntLister:    ntLister,
+				namespaces:  []string{"default"},
+				weightsName: "UserDefined",
+				ntName:      "nt-test",
+			}
+
+			// Without sleep sometimes the pods are not created in time
```

Comment:
To avoid flakiness, we usually add a wait.Poll() call to wait for the pods to be available. This is more optimal than time.Sleep().